YouTrackAutomation/Functions/New-YTIssue.ps1

```diff
@@ -0,0 +1,60 @@
+
+function New-YTIssue
+{
+    <#
+    .SYNOPSIS
+    Creates a new issue in YouTrack.
+
+    .DESCRIPTION
+    The `New-YTIssue` function creates a new issue in YouTrack. To create an issue, you must provide the project's short
+    name to the `Project` parameter, and the title of the issue to the `Summary` parameter. You can also provide an
+    optional description to the `Description` parameter.
+
+    .EXAMPLE
+    New-YTIssue -Session $session -Project 'DEMO' -Summary 'New Issue'
+
+    Demonstrates creating a new issue in the `DEMO` project with the summary `New Issue`.
+
+    .EXAMPLE
+    New-YTIssue -Session $session -Project 'DEMO' -Summary 'Write Docs' -Description 'Write YouTrackAutomation docs'
+
+    Demonstrates creating a new issue in the `DEMO` project with the summary `Write Docs` and the description `Write
+    YouTrackAutomation docs`.
+    #>
+    [CmdletBinding()]
+    param(
+        # The session object for a Youtrack Session. Create a new Session using `New-XWSession`.
+        [Parameter(Mandatory)]
+        [Object] $Session,
+
+        # The project that the issue will be created in. This should be the short name of the project.
+        [Parameter(Mandatory)]
+        [String] $Project,

```

Comment:
We want to make as few web requests as possible, so either change this to `ProjectID` or allow users to pass a project object, short name, or ID. For the second object, change its type to `Object` then create a `Resolve-YTProject` function that handles that:

```powershell
if ($Project | Get-Member -Name 'id')
{
    return $Project.id
}

if ($Project -is [int] -or $Project -match '^\d+$' )
{
    return Get-YTProject -Session $Session -ID $Project
}

if ($Project -is [String])
{
    return Get-YTProject -Session $Session -ShortName $Project
}

Write-Error "Failed to resolve project ""${Project}"": expected an object with an `id` property, an integer ID, or a project short name."
```